generate.pl

```diff
@@ -144,20 +158,22 @@ sub die_with_sample {
 
     $release->{extra_flags}    ||= '';
     $release->{debian_release} ||= 'stretch';
+    my $debug_flags = '';
 
-    if ($build =~ /main/) {
+    if ($build =~ /(main|debug)/) {
       $release->{image} = 'buildpack-deps';
       $release->{tag}   = $release->{debian_release};
+      if ($build =~ /debug/) {
+          # older versions may required -Doptimise=-g
+          $debug_flags = " -Doptimise=-g -DEBUGGING=both";
```

Comment:
```suggestion
          $debug_flags = '-Doptimise=-g -DEBUGGING=both';
```

to get rid of double-spacing

Reply:
It was in there for a reason, if I don't do that you get a whole diff for all the current files with 
```
-    && ./Configure -Darchname="$gnuArch" "$archFlag" -Dusethreads -Duseshrplib -Dvendorprefix=/usr/local  -des \
+    && ./Configure -Darchname="$gnuArch" "$archFlag" -Dusethreads -Duseshrplib -Dvendorprefix=/usr/local   -des \
```
This was a way to keep the patch smaller

Reply:
Ah, is that because of the match in https://github.com/Perl/docker-perl/pull/60/files#diff-e3548aa1c776817a085bdca9f319d2c2R198?  If so, its probably worth a comment here :+1:

Reply:
Fixed :)